include/mbgl/util/geometry_buffer.hpp

```diff
@@ -0,0 +1,287 @@
+#pragma once
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/geometries.hpp>
+#include <boost/geometry/geometries/point_xy.hpp>
+
+#include <rapidjson/prettywriter.h>
+#include <rapidjson/rapidjson.h>
+#include <rapidjson/stringbuffer.h>
+
+#include <mbgl/util/geometry.hpp>
+#include <mbgl/util/logging.hpp>
+#include <mbgl/util/optional.hpp>
+#include <mbgl/util/variant.hpp>
+
+#include <mapbox/geojson_impl.hpp>
+#include <mapbox/geometry/geometry.hpp>
+
+#include <cstdio>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+namespace mbgl {
+namespace {
+
+typedef double coordinate_type;
+
+typedef boost::geometry::model::d2::point_xy<coordinate_type> BoostPoint;
+typedef boost::geometry::model::multi_point<BoostPoint> BoostMultiPoint;
+typedef boost::geometry::model::linestring<BoostPoint> BoostLineString;
+typedef boost::geometry::model::multi_linestring<BoostLineString> BoostMultiLineString;
+typedef boost::geometry::model::polygon<BoostPoint> BoostPolygon;
+typedef boost::geometry::model::multi_polygon<BoostPolygon> BoostMultiPolygon;
+
+typedef boost::geometry::strategy::buffer::distance_symmetric<coordinate_type> DistanceStrategy;
+typedef boost::geometry::strategy::buffer::join_round JoinStrategy;
+typedef boost::geometry::strategy::buffer::end_round EndStrategy;
+typedef boost::geometry::strategy::buffer::point_circle CircleStrategy;
+typedef boost::geometry::strategy::buffer::side_straight SideStrategy;
+
+static const double earthRadius = 6371008.8;
+
+void generateFile(const std::string& filename, const std::string& data) {
+    FILE* fd = fopen(filename.c_str(), "wb");
+    if (fd) {
+        fwrite(data.data(), sizeof(std::string::value_type), data.size(), fd);
+        fclose(fd);
+    } else {
+        throw std::runtime_error(std::string("Failed to open file ") + filename);
```

Comment:
`if (!fd) throw..` where is it being caught? maybe the function could just return the status instead of throwing?